### DISCUSSION TITLE
Distribution System State Estimation

### DISCUSSION THREAD

**Discussion by jlara6** [e2nIEE/pandapower#1063] [General]
Hello, I have been reviewing the information of state estimation, as well as reviewing more bibliography of the subject, but I have some questions:

- Can I enter measurements by active or reactive power, voltage, current per phase?
- Is there an example or tutorial of state estimation for a distribution system?

**Comment by wangzhenassd** [comment]
Hi, to your question 1: asymmetrical state estimation is not supported.
to your question 2: I think, if only symmetrical state estimation is considered, it is the same as for the transmission system.

**Reply by jlara6** [reply]
Has the state estimation algorithm been tested in symmetric distribution systems?

**Reply by wangzhenassd** [reply]
I think not, WLS is in nature not suitable for MV/LV distribution grid with low measurement density

**Reply by jlara6** [reply]
If my doctoral thesis addresses the issue of state estimation in distribution systems, within the state of the art is the use of WLS and its modifications. However, I did some testing with pandapower's algorithm, but I don't understand the flowchart of the algorithm to guide me

Could you share the flowchart of the algorithm or other information that allows me to understand the architecture and the programming logic?

**Reply by wangzhenassd** [reply]
Yes, sure, basically the measurement defined in pandapower will be converted to the ppci dataformat, which contains only topological connected elements as well as measurement on bus and branches.
[https://github.com/e2nIEE/pandapower/blob/develop/pandapower/estimation/ppc_conversion.py](url)
In this file there is an extended ppci object, which contains the values required in vector form. 
Diverse algorithms are then available in algorithm directory.
If you are planning to improve these algorithms, PRs are very welcome!
